modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -77,7 +81,16 @@ object ConsensusManager {
           def getObservedOutcome: F[ConsensusOutcome[Key, Artifact]] = for {
             readyPeers <- clusterStorage.getResponsivePeers
               .map(_.filter(_.state === Ready))
-            selectedPeer <- Random[F].elementOf(readyPeers)
+            numberOfPeersToQuery <- Random[F].nextIntBounded(readyPeers.size / 4 + 1)
+            peersToQuery <- Random[F]
+              .shuffleList(readyPeers.toList)
+              .map(
```

Comment:
See @TheMMaciek comment about multiline formatting. It's better to use { instead of ( in that case.

Reply:
Changed.